rskj-core/src/main/java/co/rsk/trie/Trie.java

```diff
@@ -503,6 +507,10 @@ public Trie delete(String key) {
      * @return a byte array with the serialized info
      */
     public byte[] toMessage() {
+        if (encoded != null) {
+            return cloneArray(encoded);
```

Comment:
Doesn't cloning affect performance ? Trie.gets() also clone. Can't we just tag immutable arrays with large banners in comments and trust humans to follow the specified semantic ?

Reply:
I don't think we can trust humans. We plan to take the Pantheon approach and have immutable objects that can be safely passed around (`BytesValue`), but we can't do it until now without interfering with other humans' schedules.

See: https://github.com/PegaSysEng/pantheon/blob/master/util/src/main/java/tech/pegasys/pantheon/util/bytes/BytesValue.java